### PR TITLE
Balder/also handle non overlapping bitvectors

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge @havardpe @geirst @vekterli

### CODE REVIEW

File: searchlib/src/vespa/searchlib/common/bitvector.cpp [hunk 1]
```diff
@@ -90,10 +90,15 @@ void
 BitVector::clearInterval(Index start, Index end)
 {
     clearIntervalNoInvalidation(Range(start, end));
-
     invalidateCachedCount();
 }
 
+void
+BitVector::store(Word &word, Word value) {
+    assert(!_enable_range_check && (&word >= getActiveStart()));
```
Comment (arnej27959) [inline searchlib/src/vespa/searchlib/common/bitvector.cpp:98]: Looks like this should be || also should we check end in addition to start?

**Comment by arnej27959** [comment]
merging this now as it's likely to be an improvement at least; please review after the fact.

File: searchlib/src/tests/common/bitvector/bitvector_test.cpp [hunk 2]
```diff
@@ -365,9 +362,13 @@ verifyNonOverlappingWorksAsZeroPadded(bool clear, Func func) {
     BitVector::UP right = createEveryNthBitSet(3, 2000, 100);
     EXPECT_EQUAL(CNT, left->countTrueBits());
     EXPECT_EQUAL(CNT, right->countTrueBits());
-    EXPECT_EXCEPTION(func(*left, *right), vespalib::IllegalArgumentException,
-                     "IllegalArgumentException: [1000, 1100] starts before which is not allowed currently [2000, 2100] at "
-                     "verifyInclusiveStart in /home/balder/git/vespa/searchlib/src/vespa/searchlib/common/bitvector.cpp:33");
```
Comment (baldersheim) [inline searchlib/src/tests/common/bitvector/bitvector_test.cpp]: Thanks, did not notice the full path :)

**Review by havardpe — COMMENTED** [review]
bitvector create probably has the same issue as well...

**Pull request merged**